lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -663,14 +663,18 @@ def add_network_tokenization_card(xml, payment_method, options)
             end
             name = card_holder_name(payment_method, options)
             xml.cardHolderName name if name.present?
-            xml.cryptogram payment_method.payment_cryptogram unless options[:wallet_type] == :google_pay
+            xml.cryptogram payment_method.payment_cryptogram if should_send_payment_cryptogram?(options)
             eci = eci_value(payment_method, options)
             xml.eciIndicator eci if eci.present?
           end
           add_stored_credential_options(xml, options)
         end
       end
 
+      def should_send_payment_cryptogram?(options)
+        !(wallet_type_google_pay?(options) || (wallet_type_apple_pay?(options) && options[:stored_credential][:initiator] == 'merchant'))
```

Comment:
`wallet_type` does not indicate google vs apple, it is actually a quirk of Google Pay where Google can send a payment method of type GooglePay OR they can send the raw credit card data. 

If you're trying to check for if a payment method is apple pay I would use `payment_method.source == :apple_pay` (this assumes payment_method is of type `NetworkTokenizationCreditCard`